pom.xml

```diff
@@ -103,6 +103,18 @@
         <version>3.0</version>
         <scope>test</scope>
       </dependency>
+      <!-- 
+        To fix an issue in PCT with scm-api, this plugin is included by workflow-support
+        Otherwise LegacyJobConfigMigrationMonitorMigrationTest refuses to compile due to lack of
+        jenkins/scm/impl/mock/AbstractSampleDVCSRepoRule
```

Comment:
Note: This dependency is required via GitSampleRepoRule in git-plugin.
(Maven might lacks the feature to dependencies between artifacts classified as tests).

Reply:
I could not reproduce this issue.
Running tests with `-Djenkins.version=2.222` succeeds even without those lines.

But I believe this fix is required as  `mvn dependency:tree` shows that scm-api:tests is not included dependencies.
I rather believe that this problem should have occurred with usual test execution.

Reply:
I found what happened.

git-plugin:tests >= 3.1.0 depends on `jenkins.scm.impl.mock.AbstractSampleDVCSRepoRule` in org.jenkins-ci.plugins:scm-api.
git-plugin:tests < 3.1.0 depends on `org.jenkinsci.plugins.workflow.steps.scm.AbstractSampleDVCSRepoRule` in org.jenkins-ci.plugins.workflow:workflow-scm-step.